tests/test_live.py

```diff
@@ -0,0 +1,340 @@
+
+
+def test_cannot_draw_without_workspace():
+    with raises(RuntimeError):
+        Canvas()
+
+
+def test_cannot_draw_without_layer(ws):
+    with raises(AssertionError, match="layer"):
+        c = Canvas()
+        c.append(Rect[1, 2])
+        c.draw()
+
+
+def test_create_rect(ws, canvas, cleanup):
+    layer = Layer('M1', 'drawing')
+    r = Rect[0:0.1, 0.2:0.3, layer]
+
+    canvas.append(r)
+    (rod,) = canvas.draw()
+
+    assert rod.valid
+    assert rect_equal(r, rod.db)
+
+
+def test_create_segment(ws, canvas, cleanup):
+    layer = Layer('M2', 'pin')
+    s = Segment.from_start_end(Point(0, 1), Point(10, 1), 2, layer)
+    canvas.append(s)
+    (rod,) = canvas.draw()
+
+    assert rod.valid
+    assert segment_equal(s, rod.db)
+
+
+def test_create_group(ws, canvas, cleanup):
+    r = Rect[0:0.1, 0.2:0.3, Layer('M1', 'drawing')]
+    s = Segment.from_start_end(Point(1, 1), Point(2, 1), 0.1, Layer('M2', 'pin'))
+    g = Group([r, s])
+
+    canvas.append(g)
+    (db,) = canvas.draw()
+
+    assert db.valid
+    assert rect_equal(g.bbox, db.db)
+    assert rect_equal(db.db.figs[0], r)
+    assert segment_equal(db.db.figs[1], s)
+
+
+def test_create_nested_group(ws, canvas, cleanup):
+    one = Rect[0:0.1, 0.2:0.3, Layer('M1', 'drawing')]
+    two = one.copy()
+    two.translate(left=one.right)
+    group_one = Group([one, two])
+
+    three = one.copy()
+    three.translate(top=one.bottom)
+
+    group_two = Group([group_one, three])
+
+    canvas.append(group_two)
+    (db,) = canvas.draw()
+
+    assert rect_equal(group_two.bbox, db.db)
+    assert rect_equal(db.db.figs[1], three)
+    assert rect_equal(db.db.figs[0].figs[0], one)
+    assert rect_equal(db.db.figs[0].figs[1], two)
+
+
+def test_delete_works(ws, canvas, cleanup):
+    one = Rect[0:0.1, 0.2:0.3, Layer('M1', 'drawing')]
+    group = Group([one])
+
+    canvas.append(group)
+    (db,) = canvas.draw()
+
+    rect_db = DbShape(db.db.figs[0])
+
+    rect_db.delete()
+
+    assert not db.db.figs
+    assert not rect_db.valid
+    assert db.valid
+
+
+def test_align(ws, cv, cell_cv):
+    """
+    Crosses all alignable objects inheriting from
+    figure with PR alignment for instances, align handles and maintain flag for the align method.
+    """
+    dummy_layer = Layer("M1", "drawing")
+
+    @dataclass
+    class Alignable:
+        shape: Figure
+        align_at: str
+        rect: Rect
+
+        @property
+        def align(self):
+            if self.align_at == "b_box":
+                return self.shape.b_box
+            elif self.align_at == "pr_boundary":
+                return self.shape.pr_boundary
+
+        @property
+        def b_box(self) -> BoundingBox:
+            if self.align_at == "b_box":
+                return self.shape.skill_b_box
+            elif self.align_at == "pr_boundary":
+                return self.shape.skill_pr_boundary
+
+    count = 0
+
+    @contextmanager
+    def create_inst():
+        nonlocal count, ws, cv, cell_cv
```

Comment:
`nonlocal` is only needed, when you assign to the variable (`nonlocal count`) would be sufficient.